File: Chapter 2-7-Extract Dictionary From Array/Objective-C/ExtractDictionaryFromArray/ExtractDictionaryFromArray/ViewController.m
//
//  ViewController.m
//  ExtractDictionaryFromArray
//
//  Copyright © 2015 - by Magno Urbano - All rights reserved.
//  Part of the book Troubleshooting Xcode, published by Apress
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  // Do any additional setup after loading the view, typically from a nib.
  
  NSDate *date1 = [NSDate dateWithTimeIntervalSince1970:5000];
  NSDate *date2 = [NSDate dateWithTimeIntervalSince1970:500];
  NSDate *date3 = [NSDate dateWithTimeIntervalSince1970:2500];
  
  // the dictionaries with random dates
  NSDictionary *obj1 = @{ @"name":@"car",   @"date":date1};
  NSDictionary *obj2 = @{ @"name":@"boat",  @"date":date2};
  NSDictionary *obj3 = @{ @"name":@"plane", @"date":date3};
  // the unsorted array
  NSArray *unsortedArray = @[obj1, obj2, obj3];
  
  // create a NSSortDescriptor to sort the array
  NSSortDescriptor *dateDescriptor = [NSSortDescriptor
                                      sortDescriptorWithKey:@"date" ascending:YES];
  
  NSArray *sortDescriptors = @[dateDescriptor];
  
  // sorting the array
  NSArray *sortedArray = [unsortedArray 
                          sortedArrayUsingDescriptors:sortDescriptors];
  
  // because the array is sorted, the last object is the
  // one with the most recent date
  NSDictionary *mostRecent = [sortedArray lastObject];
  
  
  
}

- (void)didReceiveMemoryWarning {
  [super didReceiveMemoryWarning];
  // Dispose of any resources that can be recreated.
}

@end

File: Chapter 2-7-Extract Dictionary From Array/Swift 2.0/TYPE 1 - NSArray of NSDictionary/ExtractDictionaryFromArray/ExtractDictionaryFromArray/ViewController.swift
//
//  ViewController.swift
//  ExtractDictionaryFromArray
//
//  Copyright © 2015 - by Magno Urbano - All rights reserved.
//  Part of the book Troubleshooting Xcode, published by Apress
//

import UIKit

class ViewController: UIViewController {

  override func viewDidLoad() {
    super.viewDidLoad()
    // Do any additional setup after loading the view, typically from a nib.
    
    
    
    // 3 random dates
    let date1 : NSDate! = NSDate(timeIntervalSince1970: 5000);
    let date2 : NSDate! = NSDate(timeIntervalSince1970: 500);
    let date3 : NSDate! = NSDate(timeIntervalSince1970: 2500);

    // * * * * * * * * * * * * * * * * * * * *
    // * * * * * * * * * * * * * * * * * * * *
    // FIRST KIND: NSARRAY OF NSDICTIONARIES
    // * * * * * * * * * * * * * * * * * * * *
    // * * * * * * * * * * * * * * * * * * * *
    // * * * * * * * * * * * * * * * * * * * *
    // create the NSDictionaries of random NSDates
    
    let obj1 : NSDictionary? = ["name":"car",   "date":date1!]
    let obj2 : NSDictionary? = ["name":"boat",  "date":date2!]
    let obj3 : NSDictionary? = ["name":"plane", "date":date3!]
    
    // and store them on an array
    let unsortedArray : NSArray = [obj1!, obj2!, obj3!]
    
    // create a NSSortDescriptor to sort the array
    let dateDescriptor : NSSortDescriptor =
    NSSortDescriptor(key: "date", ascending: true)
    
    // sorting the array
    let sortedArrayOfDics : NSArray =  
    unsortedArray.sortedArrayUsingDescriptors([dateDescriptor])
    
    // because the array is sorted, the last object is the
    // one with the most recent date
    let mostRecent : AnyObject! = sortedArrayOfDics.lastObject
    
    
    
    
  }

  override func didReceiveMemoryWarning() {
    super.didReceiveMemoryWarning()
    // Dispose of any resources that can be recreated.
  }


}

